nixos/modules/services/monitoring/netdata.nix

```diff
@@ -329,83 +336,66 @@ in
         }
         // lib.optionalAttrs (!cfg.enableAnalyticsReporting) {
           DO_NOT_TRACK = "1";
+        }
+        // lib.optionalAttrs (cfg.claimToken != null) {
+          NETDATA_CLAIM_TOKEN = cfg.claimToken;
```

Comment:
Thank you.
I believe this is generally discouraged, anything in here will make it into the world readable nix store.


> `systemd.services.netdata.[..].LoadCredential = [ "NETDATA_CLAIM_TOKEN=<claimTokenFile> ]`


https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20loadcredential&type=code

Did you have a look for this functionality? That way you could retain the `claimTokenFile` altogether.

Reply:
OK. To answer my own question, `LoadCredential` (which was already used for the previous pre-setup script) just makes a secret file accessible to the service itself through a known path.

For here I think the right way would be to set an `EnvironmentFile`. Which would require as content `NETDATA_CLAIM_TOKEN=<the token>` inside the file.

Reply:
OK now it made click, sorry for the spam.

From the section of https://systemd.io/CREDENTIALS/
```
LoadCredential=foobar:/etc/myfoobarcredential.txt
Environment=FOOBARPATH=%d/foobar
```

>  In addition to the `$CREDENTIALS_DIRECTORY` environment variable passed to the service processes the `%d` specifier in unit files resolves to the service’s credential directory.

So in nix that would be retaining from before your commit:

```nix
          LoadCredential = [
            "netdata_claim_token:${cfg.claimTokenFile}"
          ];
```

and then adding

```nix
Environment = { NETDATA_CLAIM_TOKEN = "%d/netdata_claim_token" };
```

Reply:
Did netdata deprecate the token format?

Reply:
The script is beeing deprecated https://github.com/netdata/netdata/blob/74273e64ab2fa862fd3919ef2445dadc2b9488b8/src/claim/netdata-claim.sh.in#L106
> warning "This script is deprecated and will be officially unsupported in the near future. Please either use the kickstart script with the appropriate '--claim-*' options, or directly write out the claiming configuration instead."